src/app/cli/src/commands/system_diagnose_command.rs

```diff
@@ -41,20 +43,22 @@ pub struct SystemDiagnoseCommand {
     verification_svc: Arc<dyn VerificationService>,
     container_runtime: Arc<ContainerRuntime>,
     is_in_workpace: bool,
+    run_info_dir: PathBuf,
 }
 
 impl SystemDiagnoseCommand {
     pub fn new(
         dataset_repo: Arc<dyn DatasetRepository>,
         verification_svc: Arc<dyn VerificationService>,
         container_runtime: Arc<ContainerRuntime>,
-        is_in_workpace: bool,
+        workpace_svc: &Arc<WorkspaceService>,
```

Comment:
Why reference?

Reply:
Such as I'm using it only in constructor clippy does not allow to not use reference

Reply:
It's minor, but it would have been okay to pass 2 values instead. All our commands pass values, querying services happens in the layer above that constructs commands.